server/api/cards.go

```diff
@@ -0,0 +1,388 @@
+package api
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"net/http"
+	"strconv"
+
+	"github.com/gorilla/mux"
+	"github.com/mattermost/focalboard/server/model"
+	"github.com/mattermost/focalboard/server/services/audit"
+
+	"github.com/mattermost/mattermost-server/v6/shared/mlog"
+)
+
+const (
+	defaultPage    = "0"
+	defaultPerPage = "100"
+)
+
+func (a *API) registerCardsRoutes(r *mux.Router) {
+	// Cards APIs
+	r.HandleFunc("/boards/{boardID}/cards", a.sessionRequired(a.handleCreateCard)).Methods("POST")
+	r.HandleFunc("/boards/{boardID}/cards", a.sessionRequired(a.handleGetCards)).Methods("GET")
+	r.HandleFunc("/cards/{cardID}", a.sessionRequired(a.handlePatchCard)).Methods("PATCH")
+	r.HandleFunc("/cards/{cardID}", a.sessionRequired(a.handleGetCard)).Methods("GET")
+}
+
+func (a *API) handleCreateCard(w http.ResponseWriter, r *http.Request) {
+	// swagger:operation POST /boards/{boardID}/cards createCard
+	//
+	// Creates a new card for the specified board.
+	//
+	// ---
+	// produces:
+	// - application/json
+	// parameters:
+	// - name: boardID
+	//   in: path
+	//   description: Board ID
+	//   required: true
+	//   type: string
+	// - name: Body
+	//   in: body
+	//   description: the card to create
+	//   required: true
+	//   schema:
+	//     "$ref": "#/definitions/Card"
+	// - name: disable_notify
+	//   in: query
+	//   description: Disables notifications (for bulk data inserting)
+	//   required: false
+	//   type: bool
+	//  security:
+	// - BearerAuth: []
+	// responses:
+	//   '200':
+	//     description: success
+	//     schema:
+	//       $ref: '#/definitions/Card'
+	//   default:
+	//     description: internal error
+	//     schema:
+	//       "$ref": "#/definitions/ErrorResponse"
+
+	userID := getUserID(r)
+	boardID := mux.Vars(r)["boardID"]
+
+	val := r.URL.Query().Get("disable_notify")
+	disableNotify := val == True
+
+	requestBody, err := ioutil.ReadAll(r.Body)
```

Comment:
nit: probably not for this PR, but `ioutil.ReadAll` is being deprecated in favor of `io.ReadAll` (TIL...)

https://pkg.go.dev/io/ioutil#ReadAll

Reply:
Fixed via https://github.com/mattermost/focalboard/pull/3760/commits/65055ac6f2703631b11de467eab458bf3478a3eb